script/DCA_v2.s.sol

```diff
@@ -0,0 +1,234 @@
+        if (permit2AdapterAddress.code.length == 0) {
+            UniversalPermit2Adapter permit2Adapter = new UniversalPermit2Adapter{ salt: ZERO_SALT }(PERMIT2);
+            require(permit2AdapterAddress == address(permit2Adapter), "UniversalPermit2Adapter address mismatch.");
+        }
+
+        // NFT Descriptors
+        address tokenDescriptor = deployCreate3({
+            _name: "DCAHubPositionDescriptor",
+            _salt: 'MF-DCAHubPositionDescriptor-V1',
+            _args: hex""
+        });
+
+        // Chainlink Registry
+        address chainlinkFeedRegistry;
+        // On Ethereum, we use the `ChainlinkRegistry` operated by Chainlink.
+        if (block.chainid == 1) {
+            chainlinkFeedRegistry = 0x47Fb2585D2C56Fe188D0E6ec628a38b74fCeeeDf;
+        } else {
+            chainlinkFeedRegistry = deployCreate3({
+                _name: "ChainlinkRegistry",
+                _salt: 'MF-Chainlink-Feed-Registry-V1',
+                _args: abi.encode(msig, msigArray)
+            });
+        }
+
+        // Transformers
+        deployCreate3({_name: "ERC4626Transformer", _salt: 'MF-ERC4626-Transformer-V1', _args: abi.encode(msig)});
+        deployCreate3({_name: "ProtocolTokenWrapperTransformer", _salt: 'MF-Protocol-Transformer-V1', _args: abi.encode(msig)});
+        address transformerRegistry = deployCreate3({_name: "TransformerRegistry", _salt: 'MF-Transformer-Registry-V1', _args: abi.encode(msig)});
+        if (block.chainid == 1) {
+            address stETH = 0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84;
+            deployCreate3({_name: "wstETHTransformer", _salt: 'MF-wstETH-Transformer-V1', _args: abi.encode(stETH, msig)});
+        }
+
+        // Oracles
+        address transformerOracle = deployOracles({
+            _transformerRegistry: transformerRegistry,
+            _chainlinkFeedRegistry: chainlinkFeedRegistry
+        });
+
+        // DCA V2 Core
+        address permissionsManager = deployCreate3({_name: 'DCAPermissionsManager', _salt: 'MF-DCAV2-PermissionsManager-V1', _args:  abi.encode(msig, tokenDescriptor)});
+        address timelock = deployCreate3({_name: "TimelockController", _salt: 'MF-DCAV2-Timelock-V1', _args: abi.encode(3 days, msigArray, msigArray)});
+        address dcaHub = deployCreate3({_name: "DCAHub", _salt: 'MF-DCAV2-DCAHub-V1', _args: abi.encode(msig, timelock, transformerOracle, permissionsManager)});
+
+        // Swappers
+        address swapperRegistry = deployCreate3({_name: "SwapperRegistry", _salt: 'MF-Swapper-Registry-V1', _args: abi.encode(new address[](0), new address[](0), msig, msigArray)});
+        deployCreate3({_name: 'SwapProxy', _salt: 'MF-Swap-Proxy-V1', _args: abi.encode(swapperRegistry, msig)});
+
+        // DCA V2 Periphery
+        deployCreate3({_name: 'DCAHubCompanion', _salt: 'MF-DCAV2-DCAHubCompanion-V5', _args: abi.encode(permit2AdapterAddress, permit2AdapterAddress, msig, PERMIT2)});
```

Comment:
In the original hardhat-deploy script for `DCAHubCompanion`, the first two constructor args (the swapper) were [hard-coded](https://github.com/Mean-Finance/dca-v2-periphery/blob/main/deploy/001_companion.ts#L10) as the address of the `UniversalPermit2Adapter`, so I used the `permit2AdapterAddress` instead of hard-coding its value.